### PR TITLE
chore(test/interchain): upgrade cosmos-sdk and celestia-core versions

### PR DESCRIPTION
**Pull request by tskoyo**
Updated the dependency versions in the `test/interchain` module. This resolves build conflicts caused by differing replacement directives for:

- `github.com/cosmos/cosmos-sdk`
- `github.com/tendermint/tendermint`

Changes:
- Updated `github.com/cosmos/cosmos-sdk` to `github.com/celestiaorg/cosmos-sdk v1.25.0-sdk-v0.46.16`.
- Updated `github.com/tendermint/tendermint` to `github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.44.0-tm-v0.34.35`.

Tested with `make build` and `make test` to ensure compatibility.

### CODE REVIEW

File: test/interchain/go.mod [hunk 1]
```diff
@@ -12,15 +12,14 @@ require (
...
-	cloud.google.com/go/compute/metadata v0.2.3 // indirect
+	cloud.google.com/go/compute/metadata v0.3.0 // indirect
 	cloud.google.com/go/iam v1.1.6 // indirect
 	cloud.google.com/go/storage v1.37.0 // indirect
 	cosmossdk.io/errors v1.0.1 // indirect
 	filippo.io/edwards25519 v1.0.0-rc.1 // indirect
 	github.com/99designs/go-keychain v0.0.0-20191008050251-8e49817e8af4 // indirect
 	github.com/99designs/keyring v1.2.2 // indirect
-	github.com/BurntSushi/toml v1.3.2 // indirect
+	github.com/BurntSushi/toml v1.4.1-0.20240526193622-a339e1f7089c // indirect
```
Comment (rootulp) [inline test/interchain/go.mod:22]: this version number looks suspect so I looked into why

```
$ go mod why github.com/BurntSushi/toml
go: downloading github.com/google/pprof v0.0.0-20240827171923-fa2c70bbbfe5
go: downloading github.com/cncf/xds/go v0.0.0-20240423153145-555b57ec207b
# github.com/BurntSushi/toml
github.com/celestiaorg/celestia-app/test/interchain/chainspec
github.com/strangelove-ventures/interchaintest/v6/testutil
github.com/BurntSushi/toml
```

https://github.com/strangelove-ventures/interchaintest/blob/8f0bb15278f0d3ac33dba9e330e50b5319ab2bc2/go.mod#L19

and then I tried the bumping the versions used in the replace directive and ran `go mod tidy` and got the same result so seems fine

**Review by rootulp — APPROVED** [review]
Thanks for the PR

**Review by ninabarbakadze — DISMISSED** [review]

**Review by ninabarbakadze — COMMENTED** [review]
thanks! could you please resolve conflicts and i'll reapprove!

**Comment by tskoyo** [comment]
> thanks! could you please resolve conflicts and i'll reapprove!

done!

File: test/interchain/go.mod [hunk 2]
```diff
@@ -32,11 +31,11 @@
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/bgentry/go-netrc v0.0.0-20140422174119-9fd32a8b3d3d // indirect
 	github.com/bgentry/speakeasy v0.1.0 // indirect
-	github.com/btcsuite/btcd/btcec/v2 v2.3.2 // indirect
-	github.com/celestiaorg/nmt v0.22.0 // indirect
+	github.com/btcsuite/btcd/btcec/v2 v2.3.4 // indirect
```
[the remark this answers is not in the document]
Comment (tskoyo) [reply test/interchain/go.mod:34]: Is this something to be concerned about?

[the remark this answers is not in the document]
Comment (rootulp) [reply test/interchain/go.mod:34]: hmm I think we can upgrade the docker version in a separate PR so it shouldn't block this PR.

**Review by rootulp — APPROVED** [review]
Thanks for the PR!

**Review by ninabarbakadze — APPROVED** [review]
thanks!

**Pull request merged**